calico_node/startup/startup.go

```diff
@@ -341,6 +341,22 @@ func validateIP(ipn *net.IPNet) {
 	warning("Unable to confirm IPv%d address %s is assigned to this host", ip.Version(), ip)
 }
 
+// evaluateENVBool evaluates a passed environment variable
+// Returns an boolean indicating if the parameter was defined
+// and an boolean if the argument was true or false
+// If it was false an additional log entry will be written
+
+func evaluateENVBool(envvar string, logentry string) (isDefined bool, outgoingNATEnabled bool) {
+	envValue, isEmpty := os.LookupEnv(envvar)
+	switch strings.ToLower(envValue) {
+	case "false", "0", "no", "n", "f":
+		log.Info(logentry)
+		return isEmpty, false
+	}
+
+	return isEmpty, true
```

Comment:
I think we should tweak this just a little bit.  IIUC, right now the code will return `true` even if the environment variable isn't defined.  I think a nicer interface would be that a default is passed in, and if the environment variable isn't defined the default is returned.

e.g.

```
// Returns True if the envVar is defined and set to true.
// Returns False if the envVar is defined and set to false.
// Returns def in the envVar is not defined.
func evaluateENVBool(envVar string, def bool) (bool, bool)
```

WDYT?

It would remove the need for this bit of code below:

```
if outgoingNATisDefined == false {
    outgoingNATEnabled = false
}
```